### PR TITLE
Moves most re creations to the asm-parser constructor

### PR DESCRIPTION
**Pull request by RubenRBS**
Also adds initial support for the cv_loc stuff discussed in Discord
For #4028 but does not close it as it's only minimal support

### CODE REVIEW

File: lib/parsers/asm-parser.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ export class AsmParser extends AsmRegex {
         this.labelFindMips = /[$.A-Z_a-z][\w$.]*/g;
         this.mipsLabelDefinition = /^\$[\w$.]+:/;
         this.dataDefn = /^\s*\.(string|asciz|ascii|[1248]?byte|short|x?word|long|quad|value|zero)/;
-        this.fileFind = /^\s*\.file\s+(\d+)\s+"([^"]+)"(\s+"([^"]+)")?.*/;
+        this.fileFind = /^\s*\.(?:cv_)?file\s+(\d+)\s+"([^"]+)"(\s+"([^"]+)")?.*/;
```
Comment (mattgodbolt) [inline lib/parsers/asm-parser.js:40]: while you're here can you fix #4036 on line 44 (GH won't let me comment on a unchaanged line). probably needs to be `[\dA-Z_a-z]`)

File: lib/parsers/asm-parser.js [hunk 2]
```diff
@@ -230,9 +245,9 @@ export class AsmParser extends AsmRegex {
 
     // Remove labels which do not have a definition.
     removeLabelsWithoutDefinition(asm, labelDefinitions) {
-        _.each(asm, obj => {
+        for (const obj of asm) {
```
Comment (mattgodbolt) [inline lib/parsers/asm-parser.js:248]: so nice. much better. thanks

**Review by mattgodbolt — APPROVED** [review]

File: lib/parsers/asm-parser.js (hunk 1 above)
Comment (RubenRBS) [reply lib/parsers/asm-parser.js:40]: Done! Left it as `[$.@A-Z_a-z]\w*`

File: lib/parsers/asm-parser.js [hunk 3]
```diff
@@ -37,11 +37,11 @@ export class AsmParser extends AsmRegex {
         this.labelFindMips = /[$.A-Z_a-z][\w$.]*/g;
         this.mipsLabelDefinition = /^\$[\w$.]+:/;
         this.dataDefn = /^\s*\.(string|asciz|ascii|[1248]?byte|short|x?word|long|quad|value|zero)/;
-        this.fileFind = /^\s*\.file\s+(\d+)\s+"([^"]+)"(\s+"([^"]+)")?.*/;
+        this.fileFind = /^\s*\.(?:cv_)?file\s+(\d+)\s+"([^"]+)"(\s+"([^"]+)")?.*/;
         // Opcode expression here matches LLVM-style opcodes of the form `%blah = opcode`
         this.hasOpcodeRe = /^\s*(%[$.A-Z_a-z][\w$.]*\s*=\s*)?[A-Za-z]/;
         this.instructionRe = /^\s*[A-Za-z]+/;
-        this.identifierFindRe = /[$.@A-Z_a-z][\dA-z]*/g;
+        this.identifierFindRe = /[$.@A-Z_a-z]\w*/g;
```
Comment (partouf) [inline lib/parsers/asm-parser.js:44]: \w also includes underscore _, are there situations where labels are encoded this way?

Comment (RubenRBS) [reply lib/parsers/asm-parser.js:44]: Matt asked for `[\dA-Z_a-z]` so I thought the underscore was wanted even if it did not show up in the original RE. Happy to remove it :)

Comment (partouf) [reply lib/parsers/asm-parser.js:44]: let me give it some thought

Comment (partouf) [reply lib/parsers/asm-parser.js:44]: ah A-z does actually include _ and some other stuff. Question is do we actually need it. And I can't really answer that question, let's keep it \w until it breaks something

**Comment by partouf** [comment]
not sure whats going on, but it appears to have broken tests? https://github.com/compiler-explorer/compiler-explorer/runs/8255994794?check_suite_focus=true

**Comment by mattgodbolt** [comment]
Did we work this out? that's super odd! (yeah the identifier thing should include underscores but not `\w*` I think?)

**Comment by partouf** [comment]
according to the internet _ is included in \w https://www.w3schools.com/jsref/jsref_regexp_wordchar.asp

**Pull request merged**